lib/solidus_importer/base_importer.rb

```diff
@@ -2,8 +2,17 @@
 
 module SolidusImporter
   class BaseImporter
+    attr_reader :options, :logger
+
     def initialize(options)
       @options = options
+      @logger ||= begin
+        if @options[:logger] && !@options[:logger].is_a?(Class)
+          @options[:logger]
+        else
+          (@options[:logger] || ::SolidusImporter::BaseLogger).new
+        end
```

Comment:
I think this can be a bit more linear/readable by reducing the refs to `@options[:logger]`, here's a possible way to do it:

```suggestion
        logger = @options[:logger] || ::SolidusImporter::BaseLogger
        logger = logger.new if logger.is_a?(Class)
        logger
```